src/components/layout/createSurvey.tsx

```diff
@@ -0,0 +1,137 @@
+import { useState } from "react";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Button } from "@/components/ui/button";
+import { LoaderCircle, FilePlus } from "lucide-react";
+
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
+import { formSchema, formSchemaType } from "@/lib/validation/surveyForm";
+import { toast } from "sonner";
+import { useSession } from "@/stores/authStore";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "@/lib/queryClient";
+import { createSurvey } from "@/services/surveyService";
+
+const CreateSurveyBtn = () => {
+  const [open, setOpen] = useState(false);
+  const { session } = useSession();
+  const queryClient = useQueryClient();
+
+  const form = useForm<formSchemaType>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      name: "",
+      description: "",
+    },
+  });
+
+  const mutation = useMutation({
+    mutationFn: createSurvey,
+    onSuccess: () => {
+      toast.success("New survey created successfully!");
+      queryClient.invalidateQueries({ queryKey: queryKeys.surveys.all });
+      setOpen(false);
+      form.reset();
+    },
+    onError: () => {
+      toast.error("Something went wrong! Please try again.");
+    },
+  });
+
+  const onSubmit = (values: formSchemaType) => {
```

Comment:
```suggestion
  const onSubmit = (values: FormSchemaType) => {
```

Generally we should prefer using `PascalCase` for type declarations